workflow/src/main/kotlin/com/r3/corda/sdk/token/workflow/flows/UpdateEvolvableToken.kt

```diff
@@ -1,40 +1,137 @@
 package com.r3.corda.sdk.token.workflow.flows
 
 import co.paralleluniverse.fibers.Suspendable
-import com.r3.corda.sdk.token.contracts.commands.Update
 import com.r3.corda.sdk.token.contracts.states.EvolvableTokenType
-import com.r3.corda.sdk.token.workflow.schemas.DistributionRecord
 import com.r3.corda.sdk.token.workflow.utilities.getDistributionList
 import net.corda.core.contracts.StateAndRef
-import net.corda.core.flows.FinalityFlow
-import net.corda.core.flows.FlowLogic
-import net.corda.core.flows.InitiatingFlow
-import net.corda.core.flows.StartableByRPC
+import net.corda.core.contracts.requireThat
+import net.corda.core.flows.*
+import net.corda.core.identity.AbstractParty
+import net.corda.core.identity.Party
+import net.corda.core.node.StatesToRecord
+import net.corda.core.serialization.CordaSerializable
 import net.corda.core.transactions.SignedTransaction
 import net.corda.core.transactions.TransactionBuilder
+import net.corda.core.utilities.ProgressTracker
+import net.corda.core.utilities.unwrap
 
-@InitiatingFlow
-@StartableByRPC
-class UpdateEvolvableToken(
-        val old: StateAndRef<EvolvableTokenType>,
-        val new: EvolvableTokenType
-) : FlowLogic<SignedTransaction>() {
-    @Suspendable
-    override fun call(): SignedTransaction {
-        // Create a transaction which updates the ledger with the new evolvable token.
-        // The parties listed as maintainers in the old state should be the signers.
-        val signingKeys = old.state.data.maintainers.map { it.owningKey }
-        val utx: TransactionBuilder = TransactionBuilder(notary = old.state.notary).apply {
-            addCommand(data = Update(), keys = signingKeys)
-            addInputState(old)
-            addOutputState(state = new, contract = old.state.contract)
-        }
-        // Sign the transaction. Only Concrete Parties should be used here.
-        val stx: SignedTransaction = serviceHub.signInitialTransaction(utx)
-        // Get the list of parties on the distribution list for this evolvable token.
-        val updateSubscribers: List<DistributionRecord> = getDistributionList(serviceHub, new.linearId)
-        val sessions = updateSubscribers.map { initiateFlow(it.party) }
-        // Send to all on the list. This could take a while if there are many subscribers!
-        return subFlow(FinalityFlow(transaction = stx, sessions = sessions))
+object UpdateEvolvableToken {
```

Comment:
I know in some of the flows I went for encapsulating an initiator / responder in a named object but im not sure if this is the way forward vs having flows defined as `IssueToken` `IssueTokenHandler`. It doesn't really matter too much what we name the responder flow as, as it is never invoked. You might need to refer to it in the logs / flow watching stuff though.

Reply:
Is there a published / recommended style guide? I can follow whatever style is needed.

Agreed on the naming - much easier to use `IssueToken`.

I would call the responder `IssueTokenResponder` to make it clear that it is on the response side and not another part of the initiator.

Handle: manage (a situation or problem)
Respond: (of a person) do something as a reaction to someone or something

Reply:
Split both CreateEvolvableToken and UpdateEvolvableToken in line with recommendation. All tests pass.